### ISSUE TITLE
Heap corruption on return from main()

### ISSUE THREAD

**Issue by bliedblad** [ILT-HSR/SPICE#8]
If I try to exit the console application (return from main), I get a heap corruption exception. 

Closing the application with the "X" on the command terminal seems to bypass this issue, but would like to figure out if this can be avoided.

The issue seems to come from the POCO ethernet server, but have not been able to confirm this.

Platform - 
Windows 10
Visual Studio 2015
Release build

**Comment by ghost** [comment]
Hi

Thanks for the report. I can imagine that there is problem at the moment because the original application which it was developed for was without a shutdown functionality.

Currently I've a lot of work. I try to have a look at it until end of August.

Can you tell me which version of POCO you are using?

Thanks
Lukas

**Comment by bliedblad** [comment]
Hi,

I am using POCO 1.7.3.

This was an issue for me as device I use with becomes unresponsive if you do not close the connection to it... It should really be fixed in the device code - but it is third party and I cannot change it.

At first, I tried to use a "press ESC to exit" technique - shutting down the connection when escape was pressed, then return from main(). That's where I found the issue.

As a workaround, I hooked the "X" (kill event) and set a flag telling me to "cleanup". The kill event waits for the flag to be cleared - the while loop in main() performs the cleanup, then toggles the flag, which allows application to finish closing.

I can add this functionality to the template project and send you a pull request if you like.
